File: 0011-container-with-most-water/0011-container-with-most-water.rs
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let mut max_area=0;
        let mut i=0;
        let mut j=height.len()-1;
        while i<j {
            let mini=if &height[i]>&height[j] {j} else {i};
            let diff=(j-i) as i32;
            let area=*(&height[mini])*diff;
            if area>max_area {
                max_area=area;
            }
            if &height[i]<&height[j] {
                i+=1;
            }
            else {
                j-=1;
            }
        }
        max_area
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int area = 0, maxArea = 0;
        int l = 0, r = height.size() - 1;
        
        while(l < r) {
            int mini = (height[l] >= height[r]) ? r : l;
            
            area = (r-l) * height[mini];
            if(area > maxArea) maxArea = area;
            
            if(mini == r) {
                r--;
            }
            else if(height[r] == height[l]) {
                l++;
                r--;
            }
            else {
                l++;
            }
        }
        
        return maxArea;
    }
};